src/test/resources/lib/configFile/initIDProvider-test.js

```diff
@@ -0,0 +1,159 @@
+const test = require('/lib/xp/testing');
+
+
+
+//////////////////////  Mock workaround (test.mock only works once)
+
+const configFileMocks = {};
+test.mock("/lib/configFile/configFile.js", {
+    getAllIdProviderNames: () => configFileMocks.getAllIdProviderNames(),
+    shouldAutoInit: () => configFileMocks.shouldAutoInit(),
+    getConfigForIdProvider: (idProviderName) => configFileMocks.getConfigForIdProvider(idProviderName)
+});
+const setConfigFileMocks = mocks => {
+    Object.keys(configFileMocks).forEach( key => {
+        delete configFileMocks[key];
+    });
+    Object.keys(mocks).forEach( key => {
+        configFileMocks[key] = mocks[key];
+    })
+};
+
+const beanMocks = {};
+test.mock("/lib/configFile/services/bean.js", {
+    getIdProviders: () => beanMocks.getIdProviders(),
+    createIdProvider: (params) => beanMocks.createIdProvider(params)
+});
```

Comment:
BeanMocks will always be empty. 
To the mock will also have undefined methods. 
These are never used? Since they get overwraitten on the next line? (27 - 34)

Reply:
SetConfigFileMocks and setBeanMocks are the exact same in both test methods. 
Could we just do this once here instead?

Reply:
`beanMocks` itself is always the same object, but as a workaround for the problem described in the other comments, the members are replaced for each test. So this works, and I couldn't find a better way.